### PR TITLE
pc - migrate from react-bootstrap-table-next to react-table

### PR DESCRIPTION
**Pull request by pconrad**
# Overview

In this PR, we replace the dependency on `react-boostrap-table-next` with a dependency on `react-table`.

Reasons:
*  `react-boostrap-table-next`  doesn't seem to be being maintained
*  `react-boostrap-table-next` is causing dependency issues with npm
*  `react-boostrap-table-next` forces a hard dependency on Bootstrap

We create a new component `OurTable` which mostly replaces the functionality of `BootstrapTable`, though not completely.

* For now, all columns are sortable by default
* There may be other features of the old `BootstrapTable` that we need to find replacements for, though none of those are in the current code base.

### CODE REVIEW

**Review by kheff16 — APPROVED** [review]
Looks good to me!

File: frontend/src/main/components/OurTable.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { useTable, useSortBy } from 'react-table'
+import { Table } from "react-bootstrap";
+
+export default function OurTable({ columns, data }) {
```
Comment (btk5h) [inline frontend/src/main/components/OurTable.js:5]: Is the convention going forward to prefix our components with `Our`? We don't have any guidance on component naming conventions right now so I just wanted to clarify.

**Review by btk5h — COMMENTED** [review]
Can we get a Storybook documenting the generic `OurTable` and not just the usages of it?

**Comment by pconrad** [comment]
> Can we get a Storybook documenting the generic `OurTable` and not just the usages of it?

I think that's a good plan.   But we're on a tight deadline, so I'll put in an issue for it instead of addressing it now.  It's issue #40

**Comment by pconrad** [comment]
> > Can we get a Storybook documenting the generic `OurTable` and not just the usages of it?
> 
> I think that's a good plan. But we're on a tight deadline, so I'll put in an issue for it instead of addressing it now. It's issue #40

We do have tests for it now though.

**Pull request merged**